sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -881,18 +881,20 @@ def extract_endpoints(stage):
           if transform.spec.urn == bundle_processor.DATA_INPUT_URN:
             target = transform.unique_name, only_element(transform.outputs)
             data_input[target] = pcoll_buffers[pcoll_id]
+            coder_id = pipeline_components.pcollections[
+                only_element(transform.outputs.values())].coder_id
           elif transform.spec.urn == bundle_processor.DATA_OUTPUT_URN:
             target = transform.unique_name, only_element(transform.inputs)
             data_output[target] = pcoll_id
+            coder_id = pipeline_components.pcollections[
+                only_element(transform.inputs.values())].coder_id
           else:
             raise NotImplementedError
+          data_spec = beam_fn_api_pb2.RemoteGrpcPort(coder_id=coder_id)
```

Comment:
The PCollection coder doesn't have the windowing information, you'll need to construct one using the windowing strategies coder otherwise you'll be sending the raw coder across.

Reply:
In this runner, it actually does. (This doesn't fully fix BEAM-4150, just the worker side and enough of the runner to be compatible.)

Reply:
Hmm, in that case. LGTM.